src/corva/models/base.py

```diff
@@ -5,64 +5,62 @@
 
 import pydantic.generics
 
+from corva import utils
 from corva.api import Api
 from corva.configuration import Settings
-from corva.state.redis_adapter import RedisAdapter
 from corva.state.redis_state import RedisState
 
 
-class CorvaModelConfig:
-    arbitrary_types_allowed = True
-    extra = pydantic.Extra.allow
-    validate_assignment = True
+class EventConfig:
+    extra = pydantic.Extra.ignore
+    allow_mutation = False
 
 
-class CorvaBaseModel(pydantic.BaseModel):
-    Config = CorvaModelConfig
+class CorvaBaseEvent(pydantic.BaseModel):
+    class Config(EventConfig):
+        pass
```

Comment:
I'd rather choose the explicit version of it. Inherited empty class can be confusing

Reply:
We need `EventConfig` to be shared by `CorvaBaseEvent` and `CorvaBaseGenericEvent`.

We have a few options for this.

```python
import pydantic
from typing import ClassVar


class EventConfig:
    ...


# Option 1
class CorvaBaseEvent1(pydantic.BaseModel):
    Config = EventConfig


# Option 2
class CorvaBaseEvent2(pydantic.BaseModel):
    Config: ClassVar = EventConfig  # type: ignore


# Option 3
class CorvaBaseEvent3(pydantic.BaseModel):
    class Config(EventConfig):
        pass
```

If we use `Option 1` pycharm would list Config as a parameter to model, which may be distracting.
![image](https://user-images.githubusercontent.com/72026031/113277284-3c33e000-92e9-11eb-8648-b8e37c78cac6.png).

`Option 2` and `Option 3` dont have this issue.

Which one is better?

Reply:
I liked it how you wrote it, actually. It solves the problem without extra type overhead.